### PR TITLE
Implement speed acc rebalance

### LINKED ISSUE #73: Acc rebalance for speed pp
(In collaboration with Dumii and VINXIS)

Acc scaling for speed pp right now is pretty terrible - it's linear. Also, the effect OD has on the scaling is pretty low. Things like this is what cause people to gain pp from doubletap plays on streamy EZDT maps (and this will soon be further amplified when Xexxar's speed buff goes through).

This is what me, Dumii and VINXIS has come up with. We have changed the old linear curve to a sigmoid curve. This means that everything under a certain acc threshold gets nerfed pretty harshly, while anything above gets a very slight buff. OD also has a large effect. If the OD is lower, that acc threshold becomes higher, and thus low acc plays are nerfed more. [Link to calculator in Desmos](https://www.desmos.com/calculator/v09xnfvwtf)

Here are examples of people's profiles with this change:

[Gayzmcgee - most of his plays get a slight buff as none of them are really low acc enough to get nerfed.](https://cdn.discordapp.com/attachments/292162221511802880/525983810576187392/unknown.png)
[exc - the maps in his top that get nerfed are EZHDDT plays which range from 80-90 acc.](https://cdn.discordapp.com/attachments/292162221511802880/525984111123496970/unknown.png)
[404 AimNotFound - The selected play in the screenshot is an EZDT play on a speed map with 87% acc.](https://cdn.discordapp.com/attachments/292162221511802880/525985489736433674/unknown.png)

Would be cool to hear some thoughts from the general community before we push this change further. Thank you for reading!

### PR DESCRIPTION
**Pull request by MBmasher**
Fixes #73

### CODE REVIEW

File: src/performance/osu/OsuScore.cpp [hunk 1]
```diff
@@ -160,14 +160,14 @@ void OsuScore::computeSpeedValue(const Beatmap& beatmap)
...
-                    1 + Math.Pow(
-                        Math.E, 
-                        -20f * (Accuracy() +
-				Math.Pow(beatmap.DifficultyAttribute(_mods, Beatmap::OD), 2.0f) / 2310.0f
-				- 0.8733f
-                    )))) / 1.89f + Math.Pow(beatmap.DifficultyAttribute(_mods, Beatmap::OD), 2.0f) / 5000.0f + 0.49f;
+	_speedValue *= (
+		1.0f / (
+			1.0f + pow(
+				exp(1.0f),
```
[the remark this answers is not in the document]
Comment (MBmasher) [reply src/performance/osu/OsuScore.cpp]: how do i get the value of e without using exp(1)?

**Comment by Tom94** [comment]
You don’t need the value of e. ‘exp(x)’ already returns e^x. Your code gets e via ‘exp(1)’, which is nothing other than e^1.

**Comment by MBmasher** [comment]
Waiittttt. holy fk im an idiot

im so sorry

**Comment by smoogipoo** [comment]
@Tom94 The osu-performance peeps see this as ready.

Here's a spreadsheet of the top 200 scores, and how they change after this: https://docs.google.com/spreadsheets/d/1WiZDtYizcwPxtHPGC7R3VvZi8QS99uL81Gkt2nXlHac/edit?usp=sharing

Note: This data is compared against the master which has the FL changes merged in. Thus the spreadsheet shows the changes only relevant to this PR.

If you can look through the results and see if you agree with them, that'd be great.

**Comment by VINXIS** [comment]
Bump

also might be a good idea to use top 200 pp plays in regards to speed values instead for a more accurate representation of how plays are affected with this

**Comment by Tom94** [comment]
There still needs to be work on the implementation: it does not adhere to the coding style at all (`accLogistic` is lowercase, should not have "acc" in its name, and should be put next to other math operations within _Common.h_), has a super complex expression where the same sub-term is evaluated twice (the square of OD), and I don't see any sense in having the square of OD put into the logistic function when it _also_ additively appears outside of the logistic function.

**Comment by Tom94** [comment]
In terms of semantics, I don't think I fully grasp how the logistic function is desired here. The reasoning you claim in https://github.com/ppy/osu-performance/issues/73 talks about "nerfing low acc and buffing high acc" (same thing with OD), but a sigmoid curve (such as the logistic function) flattens _both_ the low end _and_ the high end. If your goal really was to _buff_ the high end, then you are using the wrong tool for the job, which likely explains the large number of hand-tuned magic numbers.

**Comment by VINXIS** [comment]
aaaaa we went with a sigmoid intiially with the idea of "harsh drop" in bonus via acc decrease

would [this](https://www.desmos.com/calculator/bfdvwvcai1) / an exponential function be better instead since it doesn't flatten, is much more simplified, and still keeps a harsh drop on low acc? The buff on high acc isn't meant to be a ton more than the original, and the main focus is moreso on the nature of low accuracy speed plays which is why theres a high acc intersection point for all ODs

**Comment by Tom94** [comment]
That function already looks a lot more reasonable, but I am personally still doubtful whether a simple tuning of parameters of the original formulation wouldn't have had almost exactly the same effect.

Try the original formula with `_speedValue *= 0.2f + Accuracy() * 0.8f;` instead of `_speedValue *= 0.5f + Accuracy() / 2.0f;`.

**Comment by VINXIS** [comment]
spent a bunch of time trying out stuff and yea i agree just the values need tweaking i think actually

went asking ppl and I think the best people (and i) feel is `_speedValue *= 0.02f + Accuracy();` and the od scaling to be 
`_speedValue *= 0.96f + (pow(beatmap.DifficultyAttribute(_mods, Beatmap::OD), 2) / 1600);`

**Comment by Tom94** [comment]
Awesome! I can get behind that if the data looks good. :)

**Comment by smoogipoo** [comment]
I've updated the [aforementioned sheet](https://docs.google.com/spreadsheets/d/1WiZDtYizcwPxtHPGC7R3VvZi8QS99uL81Gkt2nXlHac/edit?usp=sharing) with a new google sheet (`2019-01-08`) with the changes.

**Pull request merged**